### PR TITLE
introduce oscal cli validation and remove legacy example projects

### PR DESCRIPTION
**Pull request by wandmagic**
# Committer Notes

{Please provide a description of what this PR accomplishes. Be sure to reference any issues addressed. If the PR is a work-in-progress submitted for early review, please submit the PR as a draft PR using the "Draft pull request" dropdown.}

### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: .github/workflows/runOscalcliValidation.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Run OSCAL CLI Validation
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  workflow_dispatch:
```
Comment (david-waltermire) [inline .github/workflows/runOscalcliValidation.yml:8]: This should be setup to run on pull requests.

File: src/validations/module.mk [hunk 2]
```diff
@@ -43,3 +43,36 @@ $(VALIDATIONS_DIR)/report/test/%-junit.xml: $(VALIDATIONS_DIR)/test/%.xspec
...
+
+
+
+
+
+OSCAL_CLI_VERSION:=1.0.3
+OSCAL_CLI_BIN:=oscal-cli
+OSCAL_CLI_INSTALL_URL:=https://repo1.maven.org/maven2/gov/nist/secauto/oscal/tools/oscal-cli/cli-core/$(OSCAL_CLI_VERSION)/cli-core-$(OSCAL_CLI_VERSION)-oscal-cli.zip
+OSCAL_CLI_INSTALL_PATH:=./oscal-cli/
+OSCAL_CLI_PATH:=$(shell which $(OSCAL_CLI_BIN) > /dev/null && dirname `which $(OSCAL_CLI_BIN)` || echo $(OSCAL_CLI_INSTALL_PATH))
```
Comment (david-waltermire) [inline src/validations/module.mk:56]: Can this be setup to pull local builds? We need a way to deploy changes in PRs that haven't been merged by NIST in the metaschema-java, liboscal-java, and oscal-cli codebases.

File: .github/workflows/runOscalcliValidation.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      # Check-out submodules
+      - name: Checkout submodules
+        uses: textbook/git-checkout-submodule-action@2.1.1
+        with:
+          remote: true
+
+      # Run OSCAL CLI validation using the Makefile
+      - name: Run OSCAL CLI Validation
+        run: |
+          make validate-xml-by-cli
```
Comment (david-waltermire) [inline .github/workflows/runOscalcliValidation.yml:31]: Should the setup-java action be used to bootstrap a Java runtime?

**Review by xee5ch — COMMENTED** [review]
Hello, FedRAMP friends, helpful community member here. :wave: 

I am not sure this would help and if you want to duplicate or find a way to moduralize the logic with a Makefile approach and GHA approach, but I had created [oscal-cli-action](https://github.com/oscal-club/oscal-cli-action) for the latter for this very kind of use case. I have not done much with it due to limited community feedback, but PRs welcome. I am no sure I handle the "check for local install at path and if found just run approach," but I am more than happy to integrate that. You could probably do that with some changes to that action, and I am more than willing to accomodate with issues and PRs. Let me know.

**Comment by xee5ch** [comment]
Following up on https://github.com/GSA/fedramp-automation/pull/601#pullrequestreview-2129108508, I added https://github.com/oscal-club/oscal-cli-action/pull/10 to make how to run shell commands work, it is already supported. Let me know if you have other questions or interest in pivoting to the use of the oscal-cli action.

File: .github/workflows/runOscalcliValidation.yml (hunk 3 above)
Comment (wandmagic) [reply .github/workflows/runOscalcliValidation.yml:31]: yes, the latest commit i just tested using a local version and I was able to grab it with the dirname command

File: .github/workflows/runOscalcliValidation.yml (hunk 1 above)
Comment (wandmagic) [reply .github/workflows/runOscalcliValidation.yml]: update to run on pull requests

File: src/validations/module.mk [hunk 4]
```diff
@@ -43,3 +43,36 @@ $(VALIDATIONS_DIR)/report/test/%-junit.xml: $(VALIDATIONS_DIR)/test/%.xspec
...
+# Validate XML with oscal-cli
+#
+.PHONY: validate-xml-by-cli
+validate-xml-by-cli: $(OSCAL_CLI_PATH) ## Validate XML files by directory using OSCAL CLI Tool
+	@find $(SRC_DIR)/content/*/baselines -mindepth 1 -maxdepth 1 -type d | while read example_dir; do \
+		find "$$example_dir" -name '*.xml' | while read xml_file; do \
+			example_type=$$(echo "$$(basename $$xml_file)" | awk -F'[_.]' '{print $$(NF-1)}'); \
+			echo "Processing content type: $$example_type"; \
+			echo "Validating $$xml_file with OSCAL CLI as $$example_type"; \
+			$(OSCAL_CLI_PATH)/bin/oscal-cli "$$example_type" validate "$$xml_file"; \
```
[the remark this answers is not in the document]
Comment (wandmagic) [reply src/validations/module.mk]: added support for the other types from the dist folder but there are ALOT of errors, i think we should fix those errors and then introduce them. i have them commented out for now, but accessable from the makefile

File: src/validations/module.mk (hunk 2 above)
Comment (wandmagic) [reply src/validations/module.mk]: tested to make sure it works locally, and made some adjustments, it should work now

File: src/validations/module.mk (hunk 4 above)
[the remark this answers is not in the document]
Comment (david-waltermire) [reply src/validations/module.mk]: Are the errors comming from the oscal-cli? If so, a newer build will likely fix them. If you want, we can schedule some time on Friday to work through these.

File: .github/workflows/runOscalcliValidation.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+name: Run OSCAL CLI Validation
+
+on:
+  pull_request:
+    branches: [ main, master ]  # Adjust these branch names as needed
```
Comment (david-waltermire) [inline .github/workflows/runOscalcliValidation.yml:5]:
```suggestion
    branches:
    - main
    - develop
    - release-* # for gitflow
    - feature-* # also for gitflow
```

File: .github/workflows/runOscalcliValidation.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
```
Comment (david-waltermire) [inline .github/workflows/runOscalcliValidation.yml:21]: According to [this guide](https://whichjdk.com/#:~:text=The%20former%20AdoptOpenJDK%20project%20has,tested%20under%20a%20permissive%20license.), `temurin` is one of the better supported.

```suggestion
        distribution: 'temurin'
        cache: 'maven' # cache maven dependencies
```

File: .github/workflows/runOscalcliValidation.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+          java-version: '11'
+
+      - name: Validate XML SRC files
```
Comment (david-waltermire) [inline .github/workflows/runOscalcliValidation.yml:24]:
```suggestion
      - name: Validate SRC files
```

**Comment by david-waltermire** [comment]
@wandmagic  Can you remove the web targets and schematron validation targets from the corresponding makefiles? We won't need these going forward.

File: src/validations/module.mk (hunk 4 above)
[the remark this answers is not in the document]
Comment (wandmagic) [reply src/validations/module.mk]: the errors are coming from non-compliant constraints in the content, yes lets spend some time today to work through them

**Comment by david-waltermire** [comment]
We should merge #604 first before this PR, which will fix most of the validation issues.

**Pull request closed without merging**